public/js/actions/ArticleActions.js

```diff
@@ -0,0 +1,44 @@
+import dispatcher from '../dispatcher.js'
+
+export default class ArticleActions {
+    // TODO: Have different fetch calls for top/hot articles
+    static fetchArticles() {
+        return new Promise((resolve, reject) => {
+            fetch(`articles`)
+                .then(res => res.json())
+                .then((articles) => {
+                    articles = articles.sort((a, b) => b.votes - a.votes)
+                    dispatcher.dispatch({
+                        articles,
+                        type: 'UPDATE_ARTICLES'
```

Comment:
Since the article order was changed, the action needs to update the view correct? But it can only do that through the dispatcher to follow flux style?